src/main/java/io/getunleash/util/UnleashConfig.java

```diff
@@ -449,6 +464,10 @@ static String getDefaultInstanceId() {
             return getHostname() + "generated-" + Math.round(Math.random() * 1000000.0D);
         }
 
+        static String getDefaultConnectionId() {
+            return UUID.randomUUID().toString();
```

Comment:
this is not UUID v4 but it's the most idiomatic UUID I found in vanilla Java and we also use it for event ids.